### ISSUE TITLE
react/prop-types: Prop validation in Array.map

### ISSUE THREAD

**Issue by Lucasdsk** [yannickcr/eslint-plugin-react#2466]
![image](https://user-images.githubusercontent.com/7729844/67130134-d3c96f80-f1d6-11e9-8db8-826328a8c264.png)

In this piece of code above the eslint is warning that the `id` and `name` are missing in the props validation. But this is a destructor in a  map function. I think that this is a bug, right?

Versions:
```
eslint: 6.5.1
eslint-plugin-react: 7.16.0
```

**Comment by ljharb** [comment]
Our component detection is seeing a function that takes props and returns jsx; however it's clear that this use case shouldn't be detected as a component.

**Comment by jzabala** [comment]
This is working fine for me with versions:

eslint: ^7.3.1
eslint-plugin-react: ^7.20.2
